### DISCUSSION TITLE
Sunbird ED - Desktop app installation

### DISCUSSION THREAD

**Discussion by pushpamtrigyn** [Sunbird-Ed/Community#452] [Installation]
Hi,

I am trying to setup sunbird ed desktop app on my local machine. However, there seems to be no setup steps present under src/desktop folder in https://github.com/Sunbird-Ed/SunbirdEd-portal.

Also, let me know any specific configuration, node version, git branch or any other thing which is required.

@swayangjit @vinukumar-vs @rajeshkumaravel @anandvarada

**Comment by rajeshkumaravel** [comment]
@pushpamtrigyn 

_**Below steps and requirements is for release-5.2.0**_

### Prerequisites 

1. Node  >=14.19.0

### STEPS TO SETUP DESKTOP APP

1. Fork and clone https://github.com/Sunbird-Ed/SunbirdEd-portal
2. cd src/desktop && yarn install
3. cd OpenRAP && yarn install
4. npm run pack
5. copy path (from dist - tar file)
5. cd ..
6. yarn add path-to-tar file
7. Replace env.json file with actual env.json file (Required configuration values)
8. Copy openrap-sunbirded-plugin folder and paste
9. Open 2 terminal
    a. src/app/client ==> npm run start-desktop (let it run first then start second)
    b. src/desktop    ==> npm run start

---

CC: @chitranshu-keshav @Krishnaj20 @vinukumar-vs @rajeevsathish @swayangjit @anandvarada

**Reply by pushpamtrigyn** [reply]
Thanks @rajeshkumaravel.

For point 9 - Copy openrap-sunbirded-plugin folder and paste
Is this folder the one present in **src/desktop/modules/test_data** ? Also where exactly do I paste it ?

**Comment by rajeshkumaravel** [comment]
@pushpamtrigyn 
Place the open rap folder under `SunbirdEd-portal/src/desktop`

[openrap-sunbirded-plugin.zip](https://github.com/Sunbird-Ed/Community/files/11135021/openrap-sunbirded-plugin.zip)

CC: @chitranshu-keshav @Krishnaj20 @vinukumar-vs @rajeevsathish @swayangjit @anandvarada

**Reply by pushpamtrigyn** [reply]
@rajeshkumaravel Getting this error now, it seems it expects an ntp.json file under the openrap-sunbirded-plugin folder but it's not there.

<img width="742" alt="image" src="https://user-images.githubusercontent.com/127098111/229441476-35aaaf78-549f-4ec5-a9f8-7cd0987c5dda.png">

Also, I used this folder previously where there was some data but the file name was "sunbird.json" instead of "ntp.json" in the required path.

https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/release-5.2.0/src/desktop/modules/test_data/openrap-sunbirded-plugin

CC: @ravishtrigyn

**Reply by rajeshkumaravel** [reply]
@pushpamtrigyn 
Please verify for `CHANNEL` value under SunbirdEd-portal/src/desktop/env.json.
Please ensure that appropriate available channel is configured in the above mentioned env json file
App will try to read channel data which is configured here

CC: @chitranshu-keshav @Krishnaj20 @vinukumar-vs @rajeevsathish @swayangjit @anandvarada @ravishtrigyn

**Reply by pushpamtrigyn** [reply]
@rajeshkumaravel I have configured the channel and renamed the json file accordingly (as per the code in main.ts file). I am receiving the following error now :-

**Error: Failed to lookup view "C:\My Projects\Diksha\desktop-sunbird-ed\src\desktop\public\portal\index.ejs" in views directory "C:\My Projects\Diksha\desktop-sunbird-ed\src\desktop\views"**
    at Function.render (C:\My Projects\Diksha\desktop-sunbird-ed\src\desktop\node_modules\express\lib\application.js:597:17)
    at ServerResponse.render (C:\My Projects\Diksha\desktop-sunbird-ed\src\desktop\node_modules\express\lib\response.js:1039:7)
    at C:\My Projects\Diksha\desktop-sunbird-ed\src\desktop\modules\routes\static.ts:62:17
    at Generator.next (<anonymous>)
    at fulfilled (C:\My Projects\Diksha\desktop-sunbird-ed\src\desktop\modules\routes\static.ts:24:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

<img width="619" alt="image" src="https://user-images.githubusercontent.com/127098111/229483422-5e8ec435-1d4c-43d4-bf1f-0fedb316d000.png">

Is there supposed to be a "views" folder somewhere ? Please let me know if I am missing anything.

**Reply by pushpamtrigyn** [reply]
@rajeshkumaravel I was able to figure out the above issue, my desktop has started but it's looking something like this. Labels are not coming properly, along with some other items.

Please let me know if I could be missing anything.

<img width="937" alt="image" src="https://user-images.githubusercontent.com/127098111/229490358-b30644b7-6c29-48f1-a973-acfdb2d62e86.png">

**Reply by rajeshkumaravel** [reply]
@pushpamtrigyn 
Looks like resource bundles are not generated for desktop app.
Execute `npm run build-resource-bundles` under desktop app
The above command will execute `SunbirdEd-portal/src/desktop/resourcebundles/build.js`, which will take properties files of language keys from `SunbirdEd-portal/src/app/resourcebundles/data/consumption/en.properties` and then convert it to necessary required JSON format.
Post this activity, restart the application
You should be able to see the labels across application.

CC: @chitranshu-keshav @Krishnaj20 @vinukumar-vs @rajeevsathish @swayangjit @anandvarada @ravishtrigyn

**Reply by pushpamtrigyn** [reply]
@rajeshkumaravel `npm run build-resource-bundles` does not seem to have any effect on the desktop app.

**Reply by rajeshkumaravel** [reply]
@pushpamtrigyn 
Can you confirm whether language JSON are available under `SunbirdEd-portal/src/desktop/openrap-sunbirded-plugin/data/resourceBundles` ?

If JSON files are not present in the above folder, please copy language JSON from `SunbirdEd-portal/src/app/resourcebundles/json` and paste it under `SunbirdEd-portal/src/desktop/openrap-sunbirded-plugin/data/resourceBundles`

Restart both front end and desktop app

CC: @chitranshu-keshav @Krishnaj20 @vinukumar-vs @rajeevsathish @swayangjit @anandvarada @ravishtrigyn

**Reply by pushpamtrigyn** [reply]
@rajeshkumaravel The language JSON files are already present at `SunbirdEd-portal/src/desktop/openrap-sunbirded-plugin/data/resourceBundles`.

<img width="158" alt="Screenshot_20230404_103256" src="https://user-images.githubusercontent.com/127098111/229692883-d007ff47-1380-4c41-9b69-ad21cd1646bd.png"> 

However, I still performed a fresh copy of all JSON files from `SunbirdEd-portal/src/app/resourcebundles/json` to `SunbirdEd-portal/src/desktop/openrap-sunbirded-plugin/data/resourceBundles`, yet there seems to be no effect.

**Reply by rajeshkumaravel** [reply]
@pushpamtrigyn 
Sharing resource bundles for desktop application, please download and extract to folder `SunbirdEd-portal/src/desktop/openrap-sunbirded-plugin/data/resourceBundles`


CC: @chitranshu-keshav @Krishnaj20 @vinukumar-vs @rajeevsathish @swayangjit @anandvarada @ravishtrigyn
[resourceBundles.zip](https://github.com/Sunbird-Ed/Community/files/11145582/resourceBundles.zip)

**Reply by pushpamtrigyn** [reply]
@rajeshkumaravel It still does not work, please let me know if we can schedule a meeting to check this.

**Reply by chitranshu-keshav** [reply]
@pushpamtrigyn can you please setup a call and share the deatils with @rajeshkumaravel @rajeevsathish for them to join

cc: @Krishnaj20 @anandvarada
